xbmc/windowing/WinSystem.h

```diff
@@ -85,6 +85,9 @@ class CWinSystemBase
   virtual bool Restore() { return false; }
   virtual bool Hide() { return false; }
   virtual bool Show(bool raise = true) { return false; }
+  // Indicates if GUI layer is currently visible. When playing video it may be completely empty
+  // and can be disabled for performance reasons
+  virtual void SetVisible(bool visible) {}
```

Comment:
"can" but most likely "won't". you need to look at the codebase from an overall perspective. platform specific tweaks made the codebase rather brittle in the past.
the problem here can be solved easily without this additional interface method. if the 3rd or 4th port in the future will make use of this feature, we can bring this up again.